presto-native-execution/src/test/java/com/facebook/presto/nativeworker/AbstractTestNativeCtasQueries.java

```diff
@@ -0,0 +1,152 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.nativeworker;
+
+import com.facebook.presto.Session;
+import com.facebook.presto.testing.QueryRunner;
+import com.facebook.presto.tests.AbstractTestQueryFramework;
+import org.intellij.lang.annotations.Language;
+import org.testng.annotations.Test;
+
+import static com.facebook.presto.nativeworker.NativeQueryRunnerUtils.*;
+import static java.lang.String.format;
+import static org.testng.Assert.assertFalse;
+import static org.testng.Assert.assertTrue;
+
+public abstract class AbstractTestNativeCtasQueries
+        extends AbstractTestQueryFramework {
+
+    @Override
+    protected void createTables() {
+        QueryRunner queryRunner = (QueryRunner) getExpectedQueryRunner();
+        createLineitem(queryRunner);
+        createBucketedLineitemAndOrders(queryRunner);
+        createOrders(queryRunner);
+        createOrdersEx(queryRunner);
+        createOrdersHll(queryRunner);
+        createNation(queryRunner);
+        createCustomer(queryRunner);
+        createBucketedCustomer(queryRunner);
+        createPart(queryRunner);
+        createPartSupp(queryRunner);
+        createRegion(queryRunner);
+        createSupplier(queryRunner);
+    }
+
+    @Test
+    public void testCreateTableAsSelect() {
```

Comment:
This appears identical to the function `testCreateTableAsSelect` in `presto-tests/src/main/java/com/facebook/presto/tests/AbstractTestDistributedQueries.java`. It could be reused by adding another abstract class for CTAS tests.

Reply:
I am adding some more queries later on, so need to override it anyways.